tests/toolkit/matl_buy_policy_tests.cc

```diff
@@ -235,13 +320,187 @@ TEST_F(MatlBuyPolicyTests, ActiveDormantMultipleCycles) {
+  cyclus::Agent* a = new TestFacility(sim.context());
+  sim.context()->AddPrototype(a->prototype(), a);
+  sim.agent = sim.context()->CreateAgent<cyclus::Agent>(a->prototype());
+  sim.AddSource("commod1").Finalize();
+
+  TestFacility* fac = dynamic_cast<TestFacility*>(sim.agent);
+
+  cyclus::toolkit::ResBuf<cyclus::Material> inbuf;
+  cyclus::toolkit::MatlBuyPolicy policy;
+  policy.Init(fac, &inbuf, "inbuf", throughput, &a_dist, &d_dist, &size_dist)
+        .Set("commod1").Start();
+
+  EXPECT_NO_THROW(sim.Run());
+
+  QueryResult qr = sim.db().Query("Transactions", NULL);
+  // confirm that transactions are only occurring during active periods
+  int first_cycle = qr.GetVal<int>("Time", 0);
+  EXPECT_EQ(0, first_cycle);
+  // second cycle should star on time 8
+  int second_cycle = qr.GetVal<int>("Time", 6);
+  EXPECT_EQ(8, second_cycle);
+  // third cycle should include time 19
+  int third_cycle = qr.GetVal<int>("Time", 15);
+  EXPECT_EQ(19, third_cycle);
+
+  delete a;
+}
+
+TEST_F(MatlBuyPolicyTests, MixedActiveDormant) {
+  using cyclus::QueryResult;
+  
+  NormalIntDist a_dist = NormalIntDist(5, 1);
+  UniformIntDist d_dist = UniformIntDist(1, 3);
+  FixedDoubleDist size_dist = FixedDoubleDist(1.0);
+  
+  int dur = 12;
+  double throughput = 1;
+
+  cyclus::MockSim sim(dur);
+  cyclus::Agent* a = new TestFacility(sim.context());
+  sim.context()->AddPrototype(a->prototype(), a);
+  sim.agent = sim.context()->CreateAgent<cyclus::Agent>(a->prototype());
+  sim.AddSource("commod1").Finalize();
+
+  TestFacility* fac = dynamic_cast<TestFacility*>(sim.agent);
+
+  cyclus::toolkit::ResBuf<cyclus::Material> inbuf;
+  cyclus::toolkit::MatlBuyPolicy policy;
+  policy.Init(fac, &inbuf, "inbuf", throughput, &a_dist, &d_dist, &size_dist)
+        .Set("commod1").Start();
+
+  EXPECT_NO_THROW(sim.Run());
+
+  QueryResult qr = sim.db().Query("Transactions", NULL);
+  // confirm that transactions are only occurring during active periods
+  int first_cycle = qr.GetVal<int>("Time", 5);
+  EXPECT_EQ(5, first_cycle);
+  // second cycle should start on time 7
+  int second_cycle = qr.GetVal<int>("Time", 6);
+  EXPECT_EQ(7, second_cycle);
+
+  delete a;
+}
+
+TEST_F(MatlBuyPolicyTests, RandomSizeUniform) {
+  using cyclus::QueryResult;
+  
+  FixedIntDist a_dist = FixedIntDist(1);
+  FixedIntDist d_dist = FixedIntDist(-1);
+  UniformDoubleDist size_dist = UniformDoubleDist(0.5, 1.0);
+  
+  int dur = 2;
+  double throughput = 10;
+
+  cyclus::MockSim sim(dur);
+  cyclus::Agent* a = new TestFacility(sim.context());
+  sim.context()->AddPrototype(a->prototype(), a);
+  sim.agent = sim.context()->CreateAgent<cyclus::Agent>(a->prototype());
+  sim.AddSource("commod1").Finalize();
+
+  TestFacility* fac = dynamic_cast<TestFacility*>(sim.agent);
+
+  cyclus::toolkit::ResBuf<cyclus::Material> inbuf;
+  cyclus::toolkit::MatlBuyPolicy policy;
+  policy.Init(fac, &inbuf, "inbuf", throughput, &a_dist, &d_dist, &size_dist)
+        .Set("commod1").Start();
```

Comment:
If your default was `NULL` you could also actively just pass `NULL` to get default behavior.
```suggestion
  policy.Init(fac, &inbuf, "inbuf", throughput, NULL, NULL, &size_dist)
        .Set("commod1").Start();
```